src/main/java/org/terasology/climateConditions/igloo/IglooProvider.java

```diff
@@ -0,0 +1,102 @@
+
+package org.terasology.climateConditions.igloo;
+
+import org.terasology.core.world.CoreBiome;
+import org.terasology.core.world.generator.facets.BiomeFacet;
+import org.terasology.math.TeraMath;
+import org.terasology.math.geom.Rect2i;
+import org.terasology.math.geom.Vector2i;
+import org.terasology.utilities.procedural.Noise;
+import org.terasology.utilities.procedural.WhiteNoise;
+import org.terasology.world.generation.Border3D;
+import org.terasology.world.generation.Facet;
+import org.terasology.world.generation.FacetBorder;
+import org.terasology.world.generation.FacetProviderPlugin;
+import org.terasology.world.generation.GeneratingRegion;
+import org.terasology.world.generation.Produces;
+import org.terasology.world.generation.Requires;
+import org.terasology.world.generation.facets.SeaLevelFacet;
+import org.terasology.world.generation.facets.SurfaceHeightFacet;
+import org.terasology.world.generation.facets.base.BaseFieldFacet2D;
+import org.terasology.world.generator.plugin.RegisterPlugin;
+
+@RegisterPlugin
+@Requires({
+        @Facet(value = SurfaceHeightFacet.class, border = @FacetBorder(sides = Igloo.SIZE * 2)),
+        @Facet(value = BiomeFacet.class, border = @FacetBorder(sides = Igloo.SIZE * 2)),
+        @Facet(value = SeaLevelFacet.class, border = @FacetBorder(sides = Igloo.SIZE * 2))
+})
+@Produces(IglooFacet.class)
+public class IglooProvider implements FacetProviderPlugin {
+    private Noise noise;
+
+    @Override
+    public void setSeed(long seed) {
+        noise = new WhiteNoise(seed + 3);
+    }
+
+    /**
+     * Places the Igloo Structure in the Snow Biome with a very low probability of spawn.
+     */
+    @Override
+    public void process(GeneratingRegion region) {
+
+        Border3D border = region.getBorderForFacet(IglooFacet.class).extendBy(0, Igloo.SIZE * 2,
+                Igloo.SIZE * 2);
+
+        IglooFacet facet = new IglooFacet(region.getRegion(), border);
+        SurfaceHeightFacet surfaceHeightFacet = region.getRegionFacet(SurfaceHeightFacet.class);
+        SeaLevelFacet seaLevelFacet = region.getRegionFacet(SeaLevelFacet.class);
+        BiomeFacet biomeFacet = region.getRegionFacet(BiomeFacet.class);
+
+        Rect2i worldRegion = surfaceHeightFacet.getWorldRegion();
+
+        for (int wz = worldRegion.minY(); wz <= worldRegion.maxY(); wz++) {
+            for (int wx = worldRegion.minX(); wx <= worldRegion.maxX(); wx++) {
+                int surfaceHeight = TeraMath.floorToInt(surfaceHeightFacet.getWorld(wx, wz));
+                int seaLevel = seaLevelFacet.getSeaLevel();
+                // check if height is within this region
+                if (surfaceHeight >= facet.getWorldRegion().minY() &&
+                        surfaceHeight <= facet.getWorldRegion().maxY()) {
+                    // Sea Level + 96 is the height at which snow blocks are placed in all biomes.
+                    if (noise.noise(wx, wz) > 0.9999f && (surfaceHeight >= seaLevel + 96f ||
```

Comment:
I hope that we will get away from hard-coding snow above `sealevel + 96f` and rather place it based on climate information in the future.
To avoid "breaking" igloo generation when doing so because that "snowlevel" is still hard-coded here, I'm thinking that we should consider referencing a check for snow here and implementing this check in some more general place in `ClimateConditions`. Thus, this general place could be the source of truth of how `ClimateConditions` "looks for" snow. And then this can be changed and everything else that uses this check is automatically aligned.
Does that make sense to you?

Reply:
Umm, I'm not so sure...
No other thing in climate conditions is really checking for presence of snow anymore at the moment (maybe temperature and humidity stuff but that has a whole lot of other things which are being re-examined at the moment by RatMoleRat I think..)
And since this is the only place where we're checking snow , doing so in a separate space won't make much sense...
Also even if the snow biome is changed I do have an OR condition which checks if biome = snow. So, igloo qualifies for placement both in case height is greater than seaLevel + 96 or even if just the biome is snow......
Adding a TODO: here would make more sense in my opinion....

Reply:
Hm.. I'm still with @jdrueckert on this. Even if it's there's no one checking for snow atm, let's try to avoi localized hardcoding - for the sake of future contributors. :) Also, on line 64, could you please take the same approach for arbitraty numbers that @Cervator suggested below (for `sealevel + 96f`)?

Reply:
I'm still don't think we need to check for snow in a separate place, the issue here is the fact that : "seaLevel + 96" has been used to check if a biome is in fact the snow biome, but that has been hard coded here only because it has been done so in coreWorlds. 
I propose a constant be defined in the CoreWorlds itself which can then be used here: 
see :  https://github.com/Terasology/ClimateConditions/pull/28/files#r467621769
which would then mean another arbitrary constant won't be required here..
But I'd like to do that in a separate PR, I don't want this to get blocked form the CoreWorlds PR....
Hmmm but I think I can define an constant here now, SNOW_BIOME_THRESHOLD = seaLevel + 96
and in another PR I could update the value of the constant to a similar constant introduced in the CoreWorlds from a separate PR.
Is that fine @casals and @Cervator ?

Reply:
That's fine for me (reg. both the constant and taking the verification to another discussion/PR)